lib/src/components/NxToast/NxToast.tsx

```diff
@@ -0,0 +1,69 @@
+/*
+ * Copyright (c) 2019-present Sonatype, Inc.
+ * This program and the accompanying materials are made available under
+ * the terms of the Eclipse Public License 2.0 which accompanies this
+ * distribution and is available at https://www.eclipse.org/legal/epl-2.0/.
+ */
+import React, { useRef, useContext, useEffect, useState } from 'react';
+import classnames from 'classnames';
+
+import { nxToastPropTypes, NxToastProps } from './types';
+import NxCloseButton from '../NxCloseButton/NxCloseButton';
+import NxFontAwesomeIcon from '../NxFontAwesomeIcon/NxFontAwesomeIcon';
+
+import ToastContext from './contexts';
+import { toastTypeMap } from './toastTypeMapping';
+
+const NxToast = (props: NxToastProps) => {
+  const { toastId, className, message, type, toastContainerRef, ...otherProps } = props,
+      toastClass = toastTypeMap[type].class,
+      toastIcon = toastTypeMap[type].icon,
+      toastIconLabel = toastTypeMap[type].iconLabel,
+      [animate, setAnimate] = useState(false),
+      classes = classnames('nx-toast', 'animate', className, toastClass,
+          {'slide-in': animate},
+          {'slide-out': !animate}
+      ),
+      toastContext = useContext(ToastContext),
+      closeBtnRef = useRef<HTMLButtonElement>(null);
+
+  //Fire effect to focus on the close button of the toast that is rendered
+  useEffect(() => {
+    const closeBtn = closeBtnRef.current;
+    if (closeBtn) {
+      closeBtn.focus();
+    }
+
+    //When component is mounted, add class "animate slide-in" to trigger animation
+    setAnimate(true);
+  }, []);
+
+  //Effect to shift focus to the next toast's close button
+  //Dependency on the global toasts array
+  useEffect(() => {
+    if (toastContainerRef) {
+      //Gets the first close button of the child from the parent toast container div
+      const closeBtn = toastContainerRef.current?.querySelectorAll('.nx-btn--close')[0] as HTMLButtonElement;
```

Comment:
Is the intent to always focus the _first_ remaining toast's close button when a toast is closed?

Reply:
> Is the intent to always focus the _first_ remaining toast's close button when a toast is closed?

@JasonDSwearingen, can I get your opinion on this? Let's say there are more than two toasts stacked on top of each other. When any toast, except for the topmost one, is closed, where should the browser's focus be? The top-most toast's close button, the toast that is above the closed toast, etc...? Currently, when you close a toast among multiple toasts, focus returns to the top-most toast's close button.

Reply:
Can we have the focus move to the Toast beneath the one being dismissed (i.e. the one that moves into the space where the dismissed one was)? And if the Toast dismissed was the bottom one, the focus can move to the one above it.

Reply:
> Can we have the focus move to the Toast beneath the one being dismissed (i.e. the one that moves into the space where the dismissed one was)? And if the Toast dismissed was the bottom one, the focus can move to the one above it.

Per WW discussion, focus will go to the bottom-most toast

Reply:
> > Can we have the focus move to the Toast beneath the one being dismissed (i.e. the one that moves into the space where the dismissed one was)? And if the Toast dismissed was the bottom one, the focus can move to the one above it.
> 
> Per WW discussion, focus will go to the bottom-most toast

Now moved to focus on the bottom-most toast: [1f10ae9](https://github.com/sonatype/sonatype-react-shared-components/pull/688/commits/1f10ae908f689fad132aa82c4eb4ff75dd7aa290)

Reply:
> Now moved to focus on the bottom-most toast: [1f10ae9](https://github.com/sonatype/sonatype-react-shared-components/pull/688/commits/1f10ae908f689fad132aa82c4eb4ff75dd7aa290)

The comment above that line needs to be adjusted too

Reply:
> > Now moved to focus on the bottom-most toast: [1f10ae9](https://github.com/sonatype/sonatype-react-shared-components/pull/688/commits/1f10ae908f689fad132aa82c4eb4ff75dd7aa290)
> 
> The comment above that line needs to be adjusted too

Thanks. Updated here: [919f3cc](https://github.com/sonatype/sonatype-react-shared-components/pull/688/commits/919f3cc414b7e4d44a3471c562606f8cef37fba6)